src/DynamicDiscreteSamplers.jl

```diff
@@ -487,16 +487,17 @@ function set_global_shift_increase!(m::Memory, m2, m3::UInt64, m4) # Increase sh
     weight = UInt64(significand_sum<<shift) + 1
     when is that always 1? when
     UInt64(significand_sum<<shift) == 0
-    and because shift < 0 and significand_sum could be as much as 2^64*2^64/8/8-1 = 2^122-1,
-    shift <= -122
+    and because shift < 0 and significand_sum could be as much as 2^(Base.top_set_bit(m1))*2^64-1,
+    shift <= -Base.top_set_bit(m1)-64
     shift = signed(exponent+m3)
     shift = signed(i-4+m3)
-    signed(i-4+m3) <= -122
-    i <= -signed(m3)-122+4
-    So for -signed(m3)-118 < i, we could need to adjust the ith weight
+    signed(i-4+m3) <= -Base.top_set_bit(m1)-64
+    i <= -signed(m3)-Base.top_set_bit(m1)-64+4
+    So for -signed(m3)-60-Base.top_set_bit(m1) < i, we could need to adjust the ith weight
```

Comment:
I think it is easier to follow if we use a more precise bound in the comment:
```
    weight = UInt64(significand_sum<<shift) + 1
    when is that always 1? when
    UInt64(significand_sum<<shift) == 0
    significand_sum could be as much as m[1] * 0xfffffffffffff800. When
    shift <= -Base.top_set_bit(m[1] * 0xfffffffffffff800)
    significand_sum<<shift will be zero.
    shift = signed(exponent+m3)
    shift = signed(i-4+m3)
    signed(i-4+m3) <= -Base.top_set_bit(m[1] * 0xfffffffffffff800)
    i <= -signed(m3)-Base.top_set_bit(m[1] * 0xfffffffffffff800)+4
    So for i <= signed(m3)-Base.top_set_bit(m[1] * 0xfffffffffffff800)+4 we will not need to adjust the ith weight
    A slightly stricter and simpler condition is i <= -signed(m3)-60-Base.top_set_bit(m[1])
```

Reply:
Changed with your nicer comment